eth/stagedsync/exec3.go

```diff
@@ -764,6 +764,12 @@ Loop:
 				t1 = time.Since(tt) + ts
 
 				tt = time.Now()
+
+				// allow greedy prune on non-chain-tip
+				if err = applyTx.(state2.HasAggTx).AggTx().(*state2.AggregatorRoTx).GreedyPruneCommitHistory(ctx, applyTx, nil); err != nil {
```

Comment:
should this guarded by condition for `execStage.CurrentSyncCycle.IsInitialCycle`?

Reply:
If membatch is full then we are for sure not on chain tip